### ISSUE TITLE
Can't type `evaluate_actions` method of `ActorCriticPolicy` subclass

### ISSUE THREAD

**Issue by dfilan** [HumanCompatibleAI/imitation#578]
## Bug description
On the `cnn_scripts` branch, I've implemented [a CNN ActorCriticPolicy](https://github.com/HumanCompatibleAI/imitation/blob/cnn_scripts/src/imitation/policies/base.py#L92), and am trying to type it. Unfortunately, it seems like I can't get it to pass mypy type checking: in the SB3 `ActorCriticPolicy` [superclass](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/policies.py#L616), this method is typed as returning a tuple of three `Tensor`s. However, the third element of the returned tuple is `distribution.entropy()`. If we look at [that method](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/distributions.py#L47), it in general returns an `Optional[Tensor]`, because sometimes you can't analytically calculate the entropy of a distribution. This ends up mattering here: the `distribution` in question is returned by [`self._get_action_dist_from_latent`](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/policies.py#L581), which will sometimes return a `StateDependentNoiseDistribution`, whose `entropy` method [sometimes](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/distributions.py#L554) returns `None`. Specifically, this will happen when the `use_sde` and `squash_output` kwargs to `ActorCriticPolicy.__init__` are both set to `True`, and the action space is a `Box`.

Anyway, the upshot is that in my re-implementation, if I make the return type of the third element a `Tensor`, mypy complains `Incompatible return value type (got "Tuple[Any, Tensor, Optional[Tensor]]", expected "Tuple[Tensor, Tensor, Tensor]")`, and if I make it an `Optional[Tensor]`, I get told `Return type "Tuple[Tensor, Tensor, Optional[Tensor]]" of "evaluate_actions" incompatible with return type "Tuple[Tensor, Tensor, Tensor]" in supertype "ActorCriticPolicy"`.

Note that altho I ran into this issue when implementing a CNN, the same issue should occur whenever you want to implement an `ActorCriticPolicy` subclass that involves reimplementing the `evaluate_actions` method.

## Questions I have
- Is there a way of dealing with this situation in my code?
- Could the type signature of this function be changed in SB3? I'm actually surprised it passed typechecking. This might involve disabling the combinations of arguments that allow a `None` here.

**Comment by Rocamonde** [comment]
Based on what you're saying and on a first quick read through the code, it seems that the bug is in SB3 in that it has the wrong type annotation. 

If that is the case, I'm happy to consider submitting a PR for SB3.

**Comment by dfilan** [comment]
Yeah, not time-critical to fix this. Would be great if you could submit an SB3 PR.

**Comment by AdamGleave** [comment]
@Rocamonde @dfilan if either of you submit a PR to SB3 I'm happy to review.

Probably good to open an issue in SB3 to track this if it's a bug in the type annotation.

**Comment by Rocamonde** [comment]
Raised the issue on https://github.com/DLR-RM/stable-baselines3/issues/1117 and submitted PR on https://github.com/DLR-RM/stable-baselines3/pull/1118

**Comment by Rocamonde** [comment]
PR approved, should be merged & released soon

**Comment by dfilan** [comment]
Thanks!

**Comment by Rocamonde** [comment]
Can we close this? Or is there any additional action item?
